src/sentry/snuba/metrics/datasource.py

```diff
@@ -168,31 +193,32 @@ def get_metrics_meta(projects: Sequence[Project], use_case_id: UseCaseID) -> Seq
     metas = []
     stored_mris = get_stored_mris(projects, use_case_id) if projects else []
 
-    for mri in stored_mris:
-        parsed_mri = parse_mri(mri)
-
-        # TODO(ogi): check how is this possible
+    for metric_mri, project_ids in stored_mris.items():
+        parsed_mri = parse_mri(metric_mri)
         if parsed_mri is None:
             continue
 
-        ops = get_available_operations(parsed_mri)
-
         metas.append(
             MetricMeta(
-                mri=mri,
+                mri=metric_mri,
                 name=parsed_mri.name,
-                operations=ops,
-                # TODO: check unit casting
-                unit=parsed_mri.unit,
-                # TODO: add entity letter to entity key mapping i.e. c -> counter
-                type=parsed_mri.entity,
+                operations=cast(
+                    Sequence[MetricOperationType], get_available_operations(parsed_mri)
+                ),
+                unit=cast(MetricUnit, parsed_mri.unit),
+                type=get_entity_key_from_entity_type(
+                    parsed_mri.entity, use_case_id != UseCaseID.SESSIONS
+                ).value,
```

Comment:
Needed to revert the PR because of this:
![image](https://github.com/getsentry/sentry/assets/86684834/ceb06a9e-d028-413c-a49b-38ad6c6357ef)